framework/decode/vulkan_realign_allocator.cpp

```diff
@@ -378,10 +636,224 @@ VkResult VulkanRealignAllocator::UpdateResourceData(
+                                                  "VK_IMAGE_TILING_LINEAR image in mapped memory, "
+                                                  "playback may fail or show corruption!");
+                    }
+                    else
+                    {
+                        if (!entry->IsImageSubresourceLayoutChanged())
+                        {
+                            // There's no change with the subresource layout between capture and playback time, and
+                            // because it's linear tiling, so how texels are laid out in memory is exactly same with
+                            // capture-time. We just need to update the image data with the memory location info in
+                            // trace file plus replat-time binding offset adjustment for memory requirement change.
+                            update_data_without_change_memory_location = true;
+                        }
+                    }
+                }
+                else if (create_info.tiling == VK_IMAGE_TILING_DRM_FORMAT_MODIFIER_EXT)
+                {
+                    // It's lenear image if match the condition that Linux DRM format modifier is
+                    // DRM_FORMAT_MOD_LINEAR.
+
+                    // TODO: Add process for the following lenear image:
+                    //       VkImage created with VK_IMAGE_TILING_DRM_FORMAT_MODIFIER_EXT and whose Linux DRM format
+                    //       modifier is DRM_FORMAT_MOD_LINEAR.
+
+                    skip_update = true;
+                    GFXRECON_LOG_WARNING_ONCE("Upload data to image in mapped memory is not supported if the image "
+                                              "tiling is VK_IMAGE_TILING_DRM_FORMAT_MODIFIER_EXT!");
+                }
+            }
+
+            if (!skip_update)
+            {
+                if (update_data_without_change_memory_location)
+                {
+                    VkDeviceSize resource_start  = entry->GetTraceBindOffset();
+                    VkDeviceSize resource_end    = entry->GetTraceBindOffset() + entry->GetReplayResourceSize();
+                    uint64_t     copy_data_start = offset;
+                    uint64_t     copy_data_end   = offset + size;
+
+                    // Ignore the resource that is outside the copy range.
+                    if ((copy_data_start >= resource_end) || (resource_start >= copy_data_end))
+                    {
+                        continue;
+                    }
+
+                    VkDeviceSize copy_range_start = std::max(resource_start, copy_data_start);
+                    VkDeviceSize copy_range_end   = std::max(resource_end, copy_data_end);
+                    copy_size                     = copy_range_end - copy_range_start;
+                    mapped_memory_offset          = entry->GetReplayBindOffset() + copy_range_start - resource_start;
+                    data_offset                   = copy_range_start - copy_data_start;
```

Comment:
According to buffer case, `copy_range_end` should be `std::mix(resource_end, copy_data_end))`.

For example:
```
else if ((copy_data_start > resource_start) && (copy_data_end > resource_end))
{
    copy_size            = resource_end - copy_data_start;
```
`copy_data_start` is `std::max(resource_start, copy_data_start)`
`resource_end ` is `std::mix(resource_end, copy_data_end)`

Reply:
Okay, I will do that.

Reply:
Done.